Lib/pydoc.py

```diff
@@ -1790,7 +1790,10 @@ def doc(thing, title='Python Library Documentation: %s', forceload=0,
                 raise
             print(exc)
     else:
-        output.write(render_doc(thing, title, forceload, plaintext))
+        try:
+            output.write(render_doc(thing, title, forceload, plaintext))
+        except ImportError as exc:
+            output.write(str(exc))
```

Comment:
A general principle is to put at little in the try suite as needed to get the exception.
```suggestion
        try:
            s = render_doc(thing, title, forceload, plaintext))
        except ImportError as exc:
            s = str(exc))
        output.write(s)
```
I am going to merge with this change.